### PR TITLE
Refactor discipline incidents

### LINKED ISSUE #1809: Discipline: Total incidents and student incident counts don't update when changing time range
This was definitely working before (see GIF in https://github.com/studentinsights/studentinsights/pull/1670), but seems to have regressed.

To reproduce:
- Load the page for any school and click on the time filter.
- I was expecting the "total incidents" at the bottom left to update, and the "incidents" for each student to update.
- They seem to be unchanged regardless of the time filter.

### PR DESCRIPTION
**Pull request by edavidsonsawyer**
# What problem does this PR fix?
The discipline dashboard chart and student table were getting their data inconsistently, making the component fragile and causing bugs like #1809 

# What does this PR do?
Via suggestion from @kevinrobinson, refactors dashboard so components are using the same data. Also fixes #1809

# Screenshot (if adding a client-side feature)
goodfilter

# Checklists

## Javascript QA

+ [x] Author checked latest in IE - Discipline Dashboard
+ [x] Reviewer checked latest in IE - Discipline Dashboard

### CODE REVIEW

File: app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js [hunk 1]
```diff
@@ -43,35 +43,25 @@ class SchoolDisciplineDashboard extends React.Component {
...
-    //This chart data is filtered based on the date selection and passed to the
-    //student table and chart renders below
-    const filteredEvents = this.filterIncidentDates(this.props.schoolDisciplineEvents);
-    return {
-      type: selectedChart,
-      disciplineIncidents: _.groupBy(filteredEvents, selectedChart),
-      title: "Incidents by " + selectedChart};
-  }
-
-  sortChartKeys(chartKeys) {
```
Comment (edavidsonsawyer) [inline app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js]: This was a kludge and no longer necessary now that we have more clarity about how we want users to interact with the dashboard

File: app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js [hunk 2]
```diff
@@ -43,35 +43,25 @@ class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  studentDisciplineIncidentCounts(incidentCategory) {
+  studentDisciplineIncidentCounts(incidents) {
     let studentDisciplineIncidentCounts = {};
-    const selectedChartData = this.getChartData(this.state.selectedChart);
-    const incidents = incidentCategory ?
-                      selectedChartData.disciplineIncidents[incidentCategory] :
-                      this.filterIncidentDates(this.props.schoolDisciplineEvents);
```
Comment (edavidsonsawyer) [inline app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js]: all filtering now happens once in the render method and passed both here and the chart

File: app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js [hunk 3]
```diff
@@ -92,10 +82,10 @@ class SchoolDisciplineDashboard extends React.Component {
...
   }
 
-  sortedByIncidents(chartKeys) {
-    const chart = this.getChartData(this.state.selectedChart);
+  sortedByIncidents(groupedIncidents) {
+    const chartKeys = Object.keys(groupedIncidents);
     return chartKeys.sort((a,b) => {
-      return chart.disciplineIncidents[b].length - chart.disciplineIncidents[a].length;
+      return groupedIncidents[b].length - groupedIncidents[a].length;
     });
```
Comment (edavidsonsawyer) [inline app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js]: small change to accommodate the removal of the awkward chartData object.

File: app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js [hunk 4]
```diff
@@ -43,33 +43,25 @@ class SchoolDisciplineDashboard extends React.Component {
...
-    const filteredEvents = this.filterIncidentDates(this.props.schoolDisciplineEvents);
-    return {
-      type: selectedChart,
-      disciplineIncidents: _.groupBy(filteredEvents, selectedChart),
-      title: "Incidents by " + selectedChart};
-  }
-
-  sortChartKeys(chartKeys) {
+  sortChartKeys(groupedIncidents) {
+    const chartKeys = Object.keys(groupedIncidents)
```
Comment (kevinrobinson) [inline app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js]: This is missing a semicolon, which the linter is catching and failing the build: https://travis-ci.org/studentinsights/studentinsights/builds/393935733#L2091

Comment (kevinrobinson) [reply app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js]: @edavidsonsawyer no worries, that's what CI is for!  🚢 ing

**Comment by kevinrobinson** [comment]
verified this looks good! 👍

**Pull request merged**